diff --git a/src/nnet2/Makefile b/src/nnet2/Makefile
index 7c19ec260..8c2f26b04 100644
--- a/src/nnet2/Makefile
+++ b/src/nnet2/Makefile
@@ -21,7 +21,7 @@ OBJFILES = nnet-component.o nnet-nnet.o train-nnet.o train-nnet-ensemble.o nnet-
      get-feature-transform.o widen-nnet.o nnet-precondition-online.o \
      nnet-example-functions.o nnet-compute-discriminative.o \
      nnet-compute-discriminative-parallel.o online-nnet2-decodable.o \
-     nnet-compute-online.o
+     nnet-compute-online.o combine-nnet-a.o shrink-nnet.o

 LIBNAME = kaldi-nnet2

